### ISSUE TITLE
Random segfaults on Ubuntu 18.04

### ISSUE THREAD

**Issue by ghost** [OpenSpace/OpenSpace#1112]
OpenSpace built from commit 11775e73f56a9b4e6217576e25edb5bb24279ce3

1)
```
Thread 1 "OpenSpace" received signal SIGSEGV, Segmentation fault.
0x0000555555646a91 in glm::operator*<float, (glm::qualifier)0> (v=..., scalar=<error reading variable>)
    at /opt/OpenSpace/apps/OpenSpace/ext/sgct/include/glm/detail/type_vec4.inl:866
866			return vec<4, T, Q>(v) *= scalar;
(gdb) bt
#0  0x0000555555646a91 in glm::operator*<float, (glm::qualifier)0> (v=..., scalar=<error reading variable>)
    at /opt/OpenSpace/apps/OpenSpace/ext/sgct/include/glm/detail/type_vec4.inl:866
#1  0x000055555595b812 in glm::mat<4, 4, float, (glm::qualifier)0> glm::rotate<float, (glm::qualifier)0>(glm::mat<4, 4, float, (glm::qualifier)0> const&, float, glm::vec<3, float, (glm::qualifier)0> const&) ()
#2  0x0000555555b855f2 in openspace::RenderableSphericalGrid::update (this=0x55555ea185f0)
    at /opt/OpenSpace/modules/base/rendering/renderablesphericalgrid.cpp:301
#3  0x000055555583597c in openspace::SceneGraphNode::update (this=0x55555ea14b20, data=...) at /opt/OpenSpace/src/scene/scenegraphnode.cpp:496
#4  0x00005555557ece8f in openspace::Scene::update (this=0x55555c71eb90, data=...) at /opt/OpenSpace/src/scene/scene.cpp:321
#5  0x0000555555796f24 in openspace::RenderEngine::updateScene (this=0x555557e7b060 <openspace::global::detail::gRenderEngine()::g>)
    at /opt/OpenSpace/src/rendering/renderengine.cpp:520
#6  0x00005555556bec45 in openspace::OpenSpaceEngine::loadSingleAsset (this=0x555557e7a3e0 <openspace::global::detail::gOpenSpaceEngine()::g>, 
    assetPath="default") at /opt/OpenSpace/src/engine/openspaceengine.cpp:855
#7  0x00005555556c1408 in openspace::OpenSpaceEngine::preSynchronization (this=0x555557e7a3e0 <openspace::global::detail::gOpenSpaceEngine()::g>)
    at /opt/OpenSpace/src/engine/openspaceengine.cpp:1086
#8  0x000055555563da15 in (anonymous namespace)::mainPreSyncFunc () at /opt/OpenSpace/apps/OpenSpace/main.cpp:438
#9  0x000055555594cdfb in std::_Function_handler<void (), void (*)()>::_M_invoke(std::_Any_data const&) ()
    at /opt/OpenSpace/ext/ghoul/ext/glm/glm/detail/type_mat4x4.hpp:13
#10 0x00005555556cc0a8 in std::function<void ()>::operator()() const (this=0x555558462f40) at /usr/include/c++/8/bits/std_function.h:687
#11 0x000055555593c7fc in sgct::Engine::render() () at /opt/OpenSpace/ext/ghoul/ext/glm/glm/detail/type_mat4x4.hpp:13
#12 0x000055555564296c in main (argc=1, argv=0x7fffffffde68) at /opt/OpenSpace/apps/OpenSpace/main.cpp:1494
(gdb)
```

2)
```
Thread 1 "OpenSpace" received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
51	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007fffeed56801 in __GI_abort () at abort.c:79
#2  0x00007fffef749957 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00007fffef74fab6 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00007fffef74faf1 in std::terminate() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00007fffef74fd24 in __cxa_throw () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x00007fffef780f98 in std::__throw_ios_failure(char const*) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#7  0x00007fffef7b581a in std::basic_ios<char, std::char_traits<char> >::clear(std::_Ios_Iostate) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#8  0x00005555560d1fd8 in ghoul::opengl::ShaderObject::rebuildFromFile (this=0x5555586e2210) at /opt/OpenSpace/ext/ghoul/src/opengl/shaderobject.cpp:285
#9  0x00005555560d0bbc in ghoul::opengl::ShaderObject::ShaderObject (this=0x5555586e2210, shaderType=ghoul::opengl::ShaderObject::ShaderType::Vertex, 
    filename="/opt/OpenSpace/temp/xyuvrgba.vert", name="", dictionary=...) at /opt/OpenSpace/ext/ghoul/src/opengl/shaderobject.cpp:110
#10 0x00005555560c908d in std::make_unique<ghoul::opengl::ShaderObject, ghoul::opengl::ShaderObject::ShaderType, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, ghoul::Dictionary&> (
    __args#0=@0x7fffffffccc4: ghoul::opengl::ShaderObject::ShaderType::Vertex, __args#1="/opt/OpenSpace/temp/xyuvrgba.vert", __args#2=..., __args#3=...)
    at /usr/include/c++/8/bits/unique_ptr.h:831
#11 0x000055555609a4b3 in ghoul::opengl::ProgramObject::Build (name="xyuvrgba", vertexShaderPath="/opt/OpenSpace/temp/xyuvrgba.vert", 
    fragmentShaderPath="/opt/OpenSpace/temp/xyuvrgba.frag", dictionary=...) at /opt/OpenSpace/ext/ghoul/src/opengl/programobject.cpp:354
#12 0x000055555577b9ea in openspace::rendering::helper::initialize () at /opt/OpenSpace/src/rendering/helper.cpp:162
#13 0x00005555556bc2f4 in openspace::OpenSpaceEngine::initializeGL (this=0x555557e7a3e0 <openspace::global::detail::gOpenSpaceEngine()::g>)
    at /opt/OpenSpace/src/engine/openspaceengine.cpp:421
#14 0x000055555563cafd in (anonymous namespace)::mainInitFunc () at /opt/OpenSpace/apps/OpenSpace/main.cpp:303
#15 0x000055555594cdfb in std::_Function_handler<void (), void (*)()>::_M_invoke(std::_Any_data const&) ()
#16 0x00005555556cc0a8 in std::function<void ()>::operator()() const (this=0x555558462fc0) at /usr/include/c++/8/bits/std_function.h:687
#17 0x000055555593b0fa in sgct::Engine::initOGL() ()
#18 0x0000555555939e38 in sgct::Engine::init(sgct::Engine::RunMode) ()
#19 0x00005555556427a2 in main (argc=1, argv=0x7fffffffde68) at /opt/OpenSpace/apps/OpenSpace/main.cpp:1421
(gdb) 
```

**Comment by hv-work** [comment]
logs-glm.html for the 1st issue, log-io.html for the 2nd

[openspace-logs.zip](https://github.com/OpenSpace/OpenSpace/files/4306541/openspace-logs.zip)

**Comment by jichengtao** [comment]
The same problem 1.

**Comment by GPayne** [comment]
This appears to be fixed in new 'linux' branch, commit e6b44b5
